### PR TITLE
feat(dropdown): use semantic `<ul>` and `<li>` markup (closes #3072)

### LINKED ISSUE #3072: Switch dropdowns to use semantic <ul> and <li> markup for accessibility
Bootstrap V4 is moving back to recommending semantic markup for their dropdown menus.

Placing the dropdown items (and other sub-components) inside `<li>`  elements, and settign the dropdown menu container to a `<ul>`

https://github.com/twbs/bootstrap/pull/28591
https://github.com/twbs/bootstrap/issues/25728

We should update these components to use the new recommended markup.

### PR DESCRIPTION
**Pull request by jackmu95**
## Description of Pull Request:

> Bootstrap V4 is moving back to recommending semantic markup for their dropdown menus.
> 
> Placing the dropdown items (and other sub-components) inside `<li>` elements, and setting the dropdown menu container to a `<ul>`
> 
> twbs/bootstrap#28591
> twbs/bootstrap#25728
> 
> We should update these components to use the new recommended markup.

Closes #3072.

## PR checklist:

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [x] Enhancement to an existing feature
- [x] ARIA accessibility
- [ ] Documentation update
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e.
      `fixes #xxxx[,#xxxx]`, where "xxxx" is the issue number)
- [x] The PR should address only one issue or feature. If adding multiple features or fixing a bug
      and adding a new feature, break them into separate PRs if at all possible.
- [x] PR titles should following the
      [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e.
      "fix(alert): not alerting during SSR render", "docs(badge): Updated pill examples, fix typos",
      "chore: fix typo in docs", etc). **This is very important, as the `CHANGELOG` is generated
      from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and
      event changes)
- [ ] New/updated tests are included and passing (if required)
- [ ] Existing test suites are passing
- [ ] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (does it affect screen reader users or
      keyboard only users? clickable items should be in the tab index, etc)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a
      suggestion issue first and wait for approval before working on it)

### CODE REVIEW

**Comment by tmorehouse** [comment]
Strange.. the docs site preview isn't working right..

**Comment by tmorehouse** [comment]
Ah... nav-item-dropdown needs to have its menu container updated to a UL as well.

Might also have to tweak scrollspy a bit as well too... maybe

**Comment by jackmu95** [comment]
`<b-nav-item-dropdown>` is fixed now and the scrollspy directive works as is.

File: src/utils/object.js [hunk 1]
```diff
@@ -86,4 +86,10 @@ export const omit = (obj, props) =>
     .filter(key => props.indexOf(key) === -1)
     .reduce((result, key) => ({ ...result, [key]: obj[key] }), {})
 
+// @link https://gist.github.com/bisubus/2da8af7e801ffd813fab7ac221aa7afc
+export const pick = (obj, props) =>
+  Object.keys(obj)
```
Comment (tmorehouse) [inline src/utils/object.js]: Should use the `keys` object helper method

Also this is similar to the pluck-props utility

File: src/components/nav/nav-item-dropdown.js [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
+import { props as BDropdownProps } from '../dropdown/dropdown'
 import idMixin from '../../mixins/id'
 import dropdownMixin from '../../mixins/dropdown'
+import { pick } from '../../utils/object'
 import { htmlOrText } from '../../utils/html'
 
+// -- Constants --
+
+export const props = {
+  ...pick(BDropdownProps, ['menuClass', 'toggleClass', 'noCaret', 'role']),
```
Comment (tmorehouse) [inline src/components/nav/nav-item-dropdown.js]: Why not use the `pluckProps` method?

**Pull request merged**